packages/itwin/grouping-mapping-widget/src/components/Groups/QueryBuilder/QueryBuilder.ts

```diff
@@ -54,28 +56,43 @@ export class QueryBuilder {
   private static readonly DEFAULT_DOUBLE_PRECISION = 4;
 
   private dataProvider: PresentationPropertyDataProvider;
+  private iModelConnection: IModelConnection
   private query: Query | undefined;
 
-  constructor(provider: PresentationPropertyDataProvider) {
+  constructor(provider: PresentationPropertyDataProvider, iModelConnection: IModelConnection) {
     this.dataProvider = provider;
+    this.iModelConnection = iModelConnection
   }
 
   private isCategory(propertyField: PropertiesField): boolean {
     const classInfo = propertyField.properties[0].property.navigationPropertyInfo?.classInfo;
     return classInfo?.name === "BisCore:GeometricElement3dIsInCategory";
   }
 
+  private async getPotentialModeledElement(propertyField: PropertiesField, propertyRecord: PropertyRecord): Promise<string | undefined> {
+    const classInfo = propertyField.properties[0].property.navigationPropertyInfo?.classInfo;
+    if (propertyRecord.value?.valueFormat !== PropertyValueFormat.Primitive) return "";
+    const navigationPropertyValue = propertyRecord.value.value as NavigationPropertyValue;
+    if (classInfo?.name === "BisCore:ModelContainsElements") {
+      // Lookup the modeled element as they share the same ECInstanceId
```

Comment:
Why the modeled element and not the model itself?

Reply:
Also, I don't think there is a guarantee that modeled element and the model will share the same ECInstanceId. It would be better to use the `ModeledElement` navigation property on the Model to access the modeled element.

Reply:
https://github.com/iTwin/itwinjs-core/blob/08c0ecd13f4a01579ad1c19bbe8c1f677ecc2245/docs/bis/guide/fundamentals/model-fundamentals.md?plain=1#L43